src/AutoRest.CSharp/Common/Input/InputTypes/Transformation/InputUrlToUriTransformer.cs

```diff
@@ -2,38 +2,58 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace AutoRest.CSharp.Common.Input
 {
-    internal class InputUrlToUriTransformer
+    internal static class InputUrlToUriTransformer
     {
-        private static readonly char LowerCaseI = 'i';
-        private const string UrlSuffix = "Url";
-
         public static void Transform(InputNamespace input)
         {
             foreach (var model in input.Models)
             {
                 if (model is not InputModelType inputModel)
                     continue;
 
-                if (model.Name.AsSpan().EndsWith(UrlSuffix, StringComparison.Ordinal))
+                if (TryTransformUrlToUri(model.Name, out var newModelName))
                 {
-                    var newName = model.Name.ToCharArray().AsSpan();
-                    newName[^1] = LowerCaseI;
-                    model.Name = newName.ToString();
+                    model.Name = newModelName;
                 }
 
                 foreach (var property in inputModel.Properties)
                 {
-                    if (property.Name.AsSpan().EndsWith(UrlSuffix, StringComparison.Ordinal))
+                    if (TryTransformUrlToUri(property.Name, out var newPropertyName))
                     {
-                        var newName = model.Name.ToCharArray().AsSpan();
-                        newName[^1] = LowerCaseI;
-                        property.Name = newName.ToString();
+                        property.Name = newPropertyName;
                     }
                 }
             }
         }
+
+        internal static bool TryTransformUrlToUri(string name, [MaybeNullWhen(false)] out string newName)
+        {
+            const char i = 'i';
+            const char u = 'U';
+            const char r = 'r';
+            const char l = 'l';
+            newName = null;
+            var span = name.AsSpan();
+            if (span.Length < 3)
+            {
+                return false;
+            }
+
+            // check if this ends with `Url`
+            if (span[^3] == u && span[^2] == r && span[^1] == l)
```

Comment:
Can we still use the existing API `span.EndsWith(UrlSuffix)`? I would prefer not to rewrite the functionality if the framework already supports it. 
We can remove the `StringComparison` if not needed.

Reply:
I do not think we should remove `StringComparison`, I do not know how it is implemented but with this parameter, it is a specific overload for `ReadOnlySpan<char>` (others are taking a `T`) maybe there are optimizations only for strings performed, therefore this might be better.

Reply:
And in general, "generic code" has lower performance than "specific code".
The code I wrote here is only for checking "Url" endings which has a better performance as I showed above in the other comment.
But since the difference is minor, I changed it back.